File: algorithms/sorts/bubble-sort/bubble-sort.py
def bubble_sort(arr): 
    result_arr = arr.copy()

    for i in range(len(result_arr) - 1):
        for j in range(len(result_arr) - 1 - i):
            if result_arr[j] > result_arr[j + 1]:
                temp = result_arr[j]
                result_arr[j] = result_arr[j + 1]
                result_arr[j + 1] = temp

    return result_arr

print(bubble_sort([3, 2, 1])) # [1, 2, 3]
print(bubble_sort([4, 2, 5, 3, 1])) # [1, 2, 3, 4, 5]

File: algorithms/sorts/bubble-sort/bubble-sort.js
function bubbleSort(arr) {
  const resultArr = [...arr];

  for (let i = 0; i < resultArr.length - 1; i++) {
    for (let j = 0; j < resultArr.length - i - 1; j++) {
      if (resultArr[j] > resultArr[j + 1]) {
        const temp = resultArr[j];
        resultArr[j] = resultArr[j + 1];
        resultArr[j + 1] = temp;
      }
    }
  }

  return resultArr;
}

console.log(bubbleSort([4, 2, 5, 3, 1])); // [1, 2, 3, 4, 5]
